### ISSUE TITLE
KSType.toTypeName() is inaccurate when typealias are mixed.

### ISSUE THREAD

**Issue by nazoking** [square/kotlinpoet#1369]
When a type aliased as a generic type has a type parameter, the result of method `KSType.toTypeName()`  outputs an incorrect type name and type parameter.

The following pattern is as expected.

```kotlin
// on code,
typealias Alias2<X> = Map<X, Int>
lateinit var n3: Alias2<String>

// and in ksp processor,
resolver.getPropertyDeclarationByName("n2", true)!!.type.toTypeName().toString()
// expected:  `Alias2<String>`  , equals actual: `Alias2<String>` 
```

However, the next two patterns are incorrect

```kotlin
// on code,
typealias Alias = List<Int>
val n: List<Alias>

// and in ksp processor,
resolver.getPropertyDeclarationByName("n", true)!!.type.toTypeName().toString()
// expected:  `List<Alias>`   , but actual: `List<Alias<Int>>` 
```

```kotlin
// on code,
typealias Alias2<X> = Map<X, Int>
lateinit var n2: List<Alias2<String>>

// and in ksp processor,
resolver.getPropertyDeclarationByName("n2", true)!!.type.toTypeName().toString()
// expected:  `List<Alias2<String>>`   , but actual: `List<Alias2<String, Int>>` 
```

**Comment by ZacSweers** [comment]
What version are you using?

**Comment by nazoking** [comment]
i'm using
- kotlinpoet-ksp 1.20.0
- com.google.devtools.ksp:symbol-processing 1.7.10-1.0.6
- org.jetbrains.kotlin.jvm 1.7.10

**Comment by nazoking** [comment]
oh..?

I just ran [your test file](https://github.com/square/kotlinpoet/blob/b214a52c3e5fbc045b6daa2bf81bc16714f24716/interop/ksp/test-processor/src/test/kotlin/com/squareup/kotlinpoet/ksp/test/processor/TestProcessorTest.kt) with my test code  and run it, I get different results than what you report above. ( The results were output as expected. )
I will look into this a little more. Please let me know if you can give me any hints.
My test environment is
jdk version is 14
gradle is 7.2.

**Comment by nazoking** [comment]
ok. 

The cause was found.
The problem I faced was one that was fixed by the pull request #1321 .
This fix has not yet been released and has not been incorporated in the version 1.20.0 I am testing.
I close this Issue.
I look forward to a release in the near future.
Thanks for a great library.
